node/sync/src/block_sync.rs

```diff
@@ -874,22 +907,35 @@ impl<N: Network> BlockSync<N> {
     fn construct_requests(
         &self,
         sync_peers: &IndexMap<SocketAddr, BlockLocators<N>>,
-        current_height: u32,
+        sync_height: u32,
         min_common_ancestor: u32,
-        max_requests: usize,
+        max_blocks_to_request: u32,
     ) -> Vec<(u32, PrepareSyncRequest<N>)> {
+        // Compute the start height for the block requests.
+        let start_height = {
+            let requests = self.requests.read();
+            let mut start_height = sync_height + 1;
+
+            loop {
+                if requests.contains_key(&start_height) {
```

Comment:
Should we only skip requests which have been fulfilled? Does this relate to the redundancy factor?

Reply:
If we don't check it here it will fail on `check_next_block_request` further down. I added this for clarity and also reduce the number of log messages.

The redundancy factor is still 1 in production, so you send the request to exactly one peer.
Maybe we should change it in the future to have another timeout where after like 30 seconds you send to another peer so make progress faster?

Reply:
Thx for explaining! https://github.com/ProvableHQ/snarkOS/issues/3709